dialog_encdec.py

```diff
@@ -94,7 +112,147 @@ def gated_sent_step(self, x_t, m_t, h_tm1):
         h_t = (np.float32(1.0) - z_t) * hr_tm1 + z_t * h_tilde
          
         # return both reset state and non-reset state
-        return h_t, r_t, z_t, h_tilde
+        return h_t, r_t, z_t, h_tilde    
+
+    def gated_sent_step_pooled(self, x_t, m_t, h_tm1, h_tm1_pooled, pool_len):
```

Comment:
Cool. Perhaps one can merge this function within gated_sent_step, but by assigning default values to h_tm1_pooled and pool_len? I think scan handles this.